ddb-dm-screen.user.js

```diff
@@ -7,246 +7,201 @@
 // @match        https://www.dndbeyond.com/campaigns/*
 // @grant        none
 // ==/UserScript==
+//
 var $ = window.jQuery;
+
+class Character {
+  constructor(json) {
+    this.json = json;
+    this.stats = new Object();
+
+    this.level = 0;
+    for (var i= 0; i < json.classes.length; i++) {
+        this.level += json.classes[i].level;
+    };
+    this.proficiency = Math.ceil(this.level / 4) + 1;
+
+    var stat = ['str', 'dex', 'con', 'int', 'wis', 'cha'];
+    for(var i = 0; i < 6; i++){
+      if (json.overrideStats[i].value == null) {
+        this.stats[stat[i]] = new Stat(json.stats[i].value + json.bonusStats[i].value, this.proficiency);
+      } 
+      else {
+        this.stats[stat[i]] = new Stat(json.overrideStats[i].value, this.proficiency);
+      }
+    }
+
+    // modifiers
+    var mods = json.modifiers; 
+    [mods.race, mods.class, mods.background, mods.item, mods.feat].forEach(function(m){
+      if (m.type == "bonus") {
+        this.stats[m.subType.substr(0,3)].value += m.value;
+      }
+      else if (m.type == "proficiency") {
+        if(m.subType == 'strength-saving-throws') {
+          this.stats.str.saveProficiency = 1;
+        }
+        else {
+          this.stats[m.subType.substr(0,3)].saveProficiency = true;
+        }
+      }
+      else if (m.type == "set") {
+        this.stats[m.subType.substr(0,3)].value = m.value;
+      };
+    });
```

Comment:
How about replacing whole of this function with direct search through iframe? I wrote this below that gets stats abbreviations and it's value (but if you replace max with min you get modifier)
```
$('.ct-ability-summary').each(function(index){
  console.log($(this).find('.ct-ability-summary__label').text());
  console.log(
    Math.max(
      parseInt($(this).find('.ct-ability-summary__primary').text()),
      parseInt($(this).find('.ct-ability-summary__secondary').text())
    )
  );
});
```